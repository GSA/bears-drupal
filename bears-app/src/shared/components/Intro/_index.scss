.intro {
  padding-bottom: 30px;

  h2 {
    text-align: center;
  }

  .cta-wrapper {
    margin: 32px 32px 64px;
  }
}
