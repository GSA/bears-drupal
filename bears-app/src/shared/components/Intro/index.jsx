import PropTypes from 'prop-types'
import {
  Button,
  Chevron,
  Heading,
  NoticesList,
  ProcessList,
  TimeIndicator,
} from '../index'

/**
 * a functional component that renders a link as a button
 * @component
 * @param {object} data - inherited life event content
 * @param {object} ui - life event form ui translations
 * @param {function} setStep - incrments step count for initial form entry
 * @param {number} step - indicates which section of the form we are on
 * @return {html} returns information page view if data exist
 */
const Intro = ({ data, ui, setStep, step }) => {
  const { timeEstimate, title, summary } = data
  const { heading, timeIndicator, steps, notices, button } = ui

  return (
    data && (
      <div>
        <Chevron heading={title} description={summary} />
        <Heading headingLevel={2}>{heading}</Heading>
        <TimeIndicator
          description={timeIndicator}
          timeEstimate={timeEstimate}
        />
        <ProcessList steps={steps.list} description={steps.title} />
        <Heading headingLevel={2}>{notices.title}</Heading>
        <NoticesList data={notices.list} />
        <Button onClick={() => setStep(step + 1)}>{button}</Button>
      </div>
    )
  )
}

Intro.propTypes = {
  data: PropTypes.object,
  ui: PropTypes.object,
  setStep: PropTypes.func,
  step: PropTypes.number,
}

export default Intro
