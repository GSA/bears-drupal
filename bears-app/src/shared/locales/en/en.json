{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start",
      "list": [
        {
          "notice": "<div>Your information<strong>will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "heading": "Your potential benefits",
    "description": "<div>You must verify your eligibility and visit each agency to apply. <strong>Check with your city, state, or employer for other benefits.</strong></div>",
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    }
  },
  "notEligibleResults": {
    "heading": "Benefits you did not qualify for",
    "description": "<div>According to your answers you <strong>are not</strong> eligible for these benefits.</div>",
    "heading2": "Non-eligible Results",
    "description2": "Check with your state, city, or employer for other benefits."
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  }
}
