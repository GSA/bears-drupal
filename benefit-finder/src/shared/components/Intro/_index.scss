@use '../../styles/breakpoints/_index.scss' as *;
@use '../../styles/sizes/_index.scss' as *;
@use '../../styles/space/_index.scss' as space;

#benefit-finder {
  .intro {
    padding-bottom: space.$padding-bottom-lg;

    h2 {
      text-align: center;
    }

    .cta-wrapper {
      display: flex;
      justify-content: center;
      margin: space.$space-lg space.$space-lg space.$space-xl;
    }

    .cta-wrapper button {
      margin: 0 auto;
      width: $button-max-width;
    }

    @media (width >= $desktop) {
      .intro-process-list {
        min-width: 50%;
      }

      .intro-process-notices-heading {
        display: none;
      }

      .intro-process-group {
        display: flex;
      }
    }
  }
}
