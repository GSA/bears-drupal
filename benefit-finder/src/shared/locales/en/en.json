{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start:",
      "list": [
        {
          "notice": "<div>Your information<strong> will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "alertBanner": {
    "heading": "Error status",
    "description": "All fields marked required must be completed."
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "qualified": {
      "chevron": {
        "heading": "Your Custom List",
        "description": "<div><p><strong>Here is your list of potential Federal Benefits.</strong> Your state, city or employer may offer other benefits. Visit each agency providing the benefit to learn if you are truly eligible and conduct an actual application.</p></div>"
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you meet the criteria to qualify for these benefits:</strong></div>"
    },
    "notQualified": {
      "chevron": {
        "heading": "Benefits You Did Not Qualify For",
        "description": ""
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you did not meet the criteria to qualify for these benefits:</strong></div>"
    },
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    },
    "relativeBenefits": {
      "heading": "Other benefits that might be relevant to you"
    },
    "shareResults": {
      "heading": "Sharing and printing",
      "shareButton": "Share link",
      "emailButton": "Email results",
      "emailSubject": "Results from USAGovâ€™s benefit finder",
      "printButton": "Print results",
      "description": "Copy a link to this page with the criteria you selected above. Make sure to share it only with those you trust as your answers will be visible."
    }
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  },
  "sectionHeadings": {
    "start": "Complete Questions",
    "continue": "Questions Continued",
    "final": "Final Questions"
  },
  "select": {
    "labelSelect": "Select",
    "defaultValue": "Click to Select",
    "dateDefaultValue": "-Select-"
  },
  "date": {
    "alert": "Date should follow the format 01 - January 22 1977",
    "labelDay": "Day",
    "labelMonth": "Month",
    "labelYear": "Year",
    "monthOptions": [
      {
        "label": "01 - January"
      },
      {
        "label": "02 - February"
      },
      {
        "label": "03 - March"
      },
      {
        "label": "04 - April"
      },
      {
        "label": "05 - May"
      },
      {
        "label": "06 - June"
      },
      {
        "label": "07 - July"
      },
      {
        "label": "08 - August"
      },
      {
        "label": "09 - September"
      },
      {
        "label": "10 - October"
      },
      {
        "label": "11 - November"
      },
      {
        "label": "12 - December"
      }
    ]
  }
}
