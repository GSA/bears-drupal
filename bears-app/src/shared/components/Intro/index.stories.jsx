import Intro from './index.jsx'

export default {
  component: Intro,
  tags: ['autodocs'],
}

export const Primary = {}
