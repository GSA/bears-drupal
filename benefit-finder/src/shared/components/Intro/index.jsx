import PropTypes from 'prop-types'
import {
  Button,
  Chevron,
  Heading,
  NoticesList,
  ProcessList,
  TimeIndicator,
} from '../index'

import './_index.scss'

/**
 * a functional component that renders a link as a button
 * @component
 * @param {object} data - inherited life event content
 * @param {object} ui - life event form ui translations
 * @param {function} setStep - incrments step count for initial form entry
 * @param {number} step - indicates which section of the form we are on
 * @return {html} returns information page view if data exist
 */
const Intro = ({ data, ui, setStep, step }) => {
  const { timeEstimate, title, summary } = data
  const { heading, timeIndicator, steps, notices, button } = ui

  return (
    data && (
      <div className="intro">
        <Chevron heading={title} description={summary} />
        <div className="grid-container">
          <Heading headingLevel={2}>{heading}</Heading>
          <TimeIndicator
            description={timeIndicator}
            timeEstimate={timeEstimate}
          />
          <div className="intro-process-group">
            <div className="intro-process-list">
              <ProcessList steps={steps.list} description={steps.title} />
            </div>
            <div className="line-sperator-wrapper--vertical">
              <div className="line-sperator--vertical" />
            </div>
            <div className="intro-process-notices">
              <Heading
                className="intro-process-notices-heading"
                headingLevel={2}
              >
                {notices.heading}
              </Heading>
              <NoticesList
                className="intro-process-notices-list"
                data={notices.list}
              />
            </div>
          </div>
          <div className="line-sperator-wrapper">
            <div className="line-sperator" />
          </div>
          <div className="cta-wrapper">
            <Button onClick={() => setStep(step + 1)}>{button}</Button>
          </div>
        </div>
      </div>
    )
  )
}

Intro.propTypes = {
  data: PropTypes.object,
  ui: PropTypes.object,
  setStep: PropTypes.func,
  step: PropTypes.number,
}

export default Intro
