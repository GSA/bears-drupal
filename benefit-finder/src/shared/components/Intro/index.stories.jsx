import Intro from './index.jsx'
import content from '../../api/mock-data/current.js'
import * as en from '../../locales/en/en.json'

const { data } = JSON.parse(content)
const { lifeEventForm } = data

export default {
  component: Intro,
  parameters: {
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  args: {
    data: lifeEventForm,
    ui: en.intro,
  },
}

export const Primary = {}
