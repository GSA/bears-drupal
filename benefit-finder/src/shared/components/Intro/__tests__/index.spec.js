import renderer from 'react-test-renderer'
import Intro from '../index.jsx'
import * as en from '../../../locales/en/en.json'
import content from '../../../api/mock-data/current.js'

const t = en
const { data } = JSON.parse(content)
const { lifeEventForm } = data

describe('Intro', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(
      <Intro data={lifeEventForm} ui={t.intro} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
