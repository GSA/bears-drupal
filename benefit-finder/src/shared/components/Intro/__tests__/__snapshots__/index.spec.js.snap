// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro renders a match to the previous snapshot 1`] = `
<div
  className="intro"
>
  <div
    className="chevron"
  >
    <div
      className="grid-container"
    >
      <h1
        className=""
      >
        NEW Find federal benefits after the loss of a loved one
      </h1>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Find government benefits after the loss of a loved one including funeral, housing, and education help.</p>",
          }
        }
      />
    </div>
  </div>
  <div
    className="grid-container"
  >
    <h2
      className=""
    >
      Steps
    </h2>
    <div
      className="time-indicator"
    >
      Time to complete
       
      60 minutes
    </div>
    <div
      className="intro-process-group"
    >
      <div
        className="intro-process-list"
      >
        <ol
          className="usa-process-list"
        >
          <li
            className="usa-process-list__item"
          >
            <h4
              className="usa-process-list__heading"
            >
              Answer questions
            </h4>
          </li>
          <li
            className="usa-process-list__item"
          >
            <h4
              className="usa-process-list__heading"
            >
              Review Results
            </h4>
          </li>
          <li
            className="usa-process-list__item"
          >
            <h4
              className="usa-process-list__heading"
            >
              Find out how to apply
            </h4>
          </li>
        </ol>
      </div>
      <div
        className="intro-process-notices"
      >
        <h2
          className=""
        >
          Before you start:
        </h2>
        <div
          className="notices"
        >
          <ul
            className="notices-list add-list-reset"
          >
            <li
              className="notice"
            >
              <div
                className=""
                role="alert"
              >
                <div
                  className="usa-alert__body"
                >
                  <div
                    className="usa-alert__text"
                  >
                    <div
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<div>Your information<strong> will not</strong> be shared, saved, or submitted.</div>",
                        }
                      }
                    />
                  </div>
                </div>
              </div>
            </li>
            <li
              className="notice"
            >
              <div
                className=""
                role="alert"
              >
                <div
                  className="usa-alert__body"
                >
                  <div
                    className="usa-alert__text"
                  >
                    <div
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "Provide complete and accurate information to get the most relevant results.",
                        }
                      }
                    />
                  </div>
                </div>
              </div>
            </li>
            <li
              className="notice"
            >
              <div
                className=""
                role="alert"
              >
                <div
                  className="usa-alert__body"
                >
                  <div
                    className="usa-alert__text"
                  >
                    <div
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>",
                        }
                      }
                    />
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="line-sperator-wrapper"
    >
      <div
        className="line-sperator"
      />
    </div>
    <div
      className="cta-wrapper"
    >
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        Start
      </button>
    </div>
  </div>
</div>
`;
